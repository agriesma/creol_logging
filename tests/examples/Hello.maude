load creol-interpreter .
mod PROGRAM is
  protecting CREOL-SIMULATOR .
  op classes : -> Configuration [ctor] .
  eq classes =
    < "Console" : Class | Inh: noInh, Param: noVid, Att: "output" |-> null,
      Mtds: < "init" : Method | Param: noVid, Att: "_" |-> null,
              Code: assign( ( "output" @ "Console" ) ; list(emp) ) ;
              return ( emp ) >,
            < "run" : Method | Param: noVid, Att: "_" |-> null, Code: skip ;
              return ( emp ) >,
            < "print" : Method | Param: "result", Att: "result" |-> null, "_"
              |-> null, Code: assign( "output" ; "|-" ( "output" ::
              str("Hello, world") ) ) ; return ( emp ) >,
      Ocnt: 0 >
    
    < "Hello" : Class | Inh: noInh, Param: noVid, Att: "console" |-> null,
      Mtds: < "init" : Method | Param: noVid, Att: "_" |-> null,
              Code: new( "console" ; "Console" ; emp ) ; return ( emp ) >,
            < "run" : Method | Param: noVid, Att: "label:0" |-> null, "_" |->
              null, Code: call( "label:0" ; "console" ; "print" ;
              str("hello, world") ) ; get( "label:0" ; noVid ) ;
              free( "label:0" ) ; return ( emp ) >,
      Ocnt: 0 > .
endm

