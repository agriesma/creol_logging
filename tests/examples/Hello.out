==========================================
reduce in PROGRAM : classes .
result Configuration: 
< "Console" : Class | Inh: noInh, Param: noVid, Att: "output" |-> null, Mtds:
  < "init" : Method | Param: noVid, Att: "_" |-> null, Code: assign("output" @
    "Console" ; list(emp)) ; return(emp) >, < "print" : Method | Param:
    "result", Att: "_" |-> null, "result" |-> null, Code: assign("output" ;
    "|-"("output" :: str("Hello, world"))) ; return(emp) >, < "run" : Method |
    Param: noVid, Att: "_" |-> null, Code: skip ; return(emp) >,
  Ocnt: 0 >
 
< "Hello" : Class | Inh: noInh, Param: noVid, Att: "console" |-> null, Mtds:
  < "init" : Method | Param: noVid, Att: "_" |-> null, Code: new("console" ;
    "Console" ; emp) ; return(emp) >, < "run" : Method | Param: noVid, Att: "_"
    |-> null, "label:0" |-> null, Code: call("label:0" ; "console" ; "print" ;
    str("hello, world")) ; get("label:0" ; noVid) ; free("label:0") ; return(
    emp) >,
  Ocnt: 0 >

==========================================
rewrite in PROGRAM : main(classes, "Hello", emp) .
result State: {
< ob("Console0") : "Console" |
  Att: "output" |-> list(str("Hello, world")), "this" |-> ob("Console0"),
  Pr: idle,
  PrQ: noProc,
  Lcnt: 2 >
 
< ob("Hello0") : "Hello" |
  Att: "console" |-> ob("Console0"), "this" |-> ob("Hello0"),
  Pr: idle,
  PrQ: noProc,
  Lcnt: 4 >
 
< ob("main") : Start |
  Att: noSubst,
  Pr: idle,
  PrQ: noProc,
  Lcnt: 1 >
 
< "Console" : Class | Inh: noInh, Param: noVid, Att: "output" |-> null, Mtds:
  < "init" : Method | Param: noVid, Att: "_" |-> null, Code: assign("output" @
    "Console" ; list(emp)) ; return(emp) >, < "print" : Method | Param:
    "result", Att: "_" |-> null, "result" |-> null, Code: assign("output" ;
    "|-"("output" :: str("Hello, world"))) ; return(emp) >, < "run" : Method |
    Param: noVid, Att: "_" |-> null, Code: skip ; return(emp) >,
  Ocnt: 1 >
 
< "Hello" : Class | Inh: noInh, Param: noVid, Att: "console" |-> null, Mtds:
  < "init" : Method | Param: noVid, Att: "_" |-> null, Code: new("console" ;
    "Console" ; emp) ; return(emp) >, < "run" : Method | Param: noVid, Att: "_"
    |-> null, "label:0" |-> null, Code: call("label:0" ; "console" ; "print" ;
    str("hello, world")) ; get("label:0" ; noVid) ; free("label:0") ; return(
    emp) >,
  Ocnt: 1 >
}
Bye.
